data-products/src/main/scala/org/sunbird/analytics/sourcing/SourcingMetrics.scala

```diff
@@ -0,0 +1,173 @@
+case class ContentReportResult(identifier: String, l1identifier: String, contentType: String)
+case class TextbookHierarchy(channel: String, board: String, identifier: String, medium: Object, gradeLevel: List[String], subject: Object,
+                             name: String, status: String, contentType: Option[String], leafNodesCount: Int, lastUpdatedOn: String,
+                             depth: Int, createdOn: String, children: Option[List[TextbookHierarchy]], index: Int, parent: String)
+case class FinalReport(identifier: String,l1identifier: String,board: String, medium: String, grade: String, subject: String, name: String, chapters: String, channel: String, totalChapters: String, slug:String)
+case class TextbookResponse(l1identifier:String,board: String, medium: String, grade: String, subject: String, name: String, chapters: String, channel: String)
+
+object SourcingMetrics extends optional.Application with IJob with BaseReportsJob {
+
+  implicit val className = "org.sunbird.analytics.job.SourcingMetrics"
+  val jobName: String = "Sourcing Metrics Job"
+  val sunbirdHierarchyStore: String = AppConf.getConfig("course.metrics.cassandra.sunbirdHierarchyStore")
+
+  // $COVERAGE-OFF$ Disabling scoverage for main method
+  def main(config: String)(implicit sc: Option[SparkContext], fc: Option[FrameworkContext]): Unit = {
+    JobLogger.log(s"Started execution - $jobName",None, Level.INFO)
+    implicit val sparkContext: SparkContext = getReportingSparkContext(JSONUtils.deserialize[JobConfig](config))
+    implicit val frameworkContext: FrameworkContext = getReportingFrameworkContext()
+    val readConsistencyLevel: String = AppConf.getConfig("course.metrics.cassandra.input.consistency")
+    val sparkConf = sparkContext.getConf
+      .set("es.write.operation", "upsert")
+      .set("spark.cassandra.input.consistency.level", readConsistencyLevel)
+    val spark = SparkSession.builder.config(sparkConf).getOrCreate()
+
+    generateSourcingMetrics(spark, config)
+  }
+
+  // $COVERAGE-ON$ Enabling scoverage for all other functions
+  def generateSourcingMetrics(spark: SparkSession, config: String)(implicit sc: SparkContext,fc: FrameworkContext): Unit = {
+    val conf = JSONUtils.deserialize[Map[String,AnyRef]](config)
+    val textbooks = TextBookUtils.getTextBooks(conf, RestUtil)
+    JobLogger.log(s"Fetched textbooks from druid ${textbooks.length}",None, Level.INFO)
+    import spark.implicits._
+
+    val textbookReportData = getTextbookInfo(textbooks, spark, config)
+    val textbookReports = textbookReportData._1.toDF()
+    val tenantInfo = getTenantInfo(RestUtil).toDF()
+    val report = textbookReports.join(tenantInfo,
+      textbookReports.col("channel") === tenantInfo.col("id"),"left")
+    generateSourcingReport(spark, textbookReportData._2.toDF(), report, config)
+  }
+
+  def generateSourcingReport(spark: SparkSession, contentdf: DataFrame, report: DataFrame, config: String): Unit = {
```

Comment:
SparkSession, FrameworkContext and JobConfig should be passed as implicit parameters